metadata-ingestion/scripts/docgen.py

```diff
@@ -529,228 +273,75 @@ def generate(
                         # all other docs are assumed to be plugin level
-                        create_or_update(
-                            source_documentation,
-                            [platform_name, "custom_docs"],
-                            final_markdown,
-                        )
-                    else:
-                        if "_" in file_name:
-                            plugin_doc_parts = file_name.split("_")
-                            if len(plugin_doc_parts) != 2 or plugin_doc_parts[
-                                1
-                            ] not in ["pre", "post"]:
-                                raise Exception(
-                                    f"{file_name} needs to be of the form <plugin>_pre.md or <plugin>_post.md"
-                                )
-
-                            docs_key_name = f"custom_docs_{plugin_doc_parts[1]}"
-                            create_or_update(
-                                source_documentation,
-                                [
-                                    platform_name,
-                                    "plugins",
-                                    plugin_doc_parts[0],
-                                    docs_key_name,
-                                ],
-                                final_markdown,
+                        platforms[platform_name].custom_docs_pre = final_markdown
+
+                    elif "_" in file_name:
+                        plugin_doc_parts = file_name.split("_")
+                        if len(plugin_doc_parts) != 2:
+                            raise ValueError(
+                                f"{file_name} needs to be of the form <plugin>_pre.md or <plugin>_post.md"
                             )
+                        plugin_name, suffix = plugin_doc_parts
+                        if suffix == "pre":
+                            platforms[platform_name].plugins[
+                                plugin_name
+                            ].custom_docs_pre = final_markdown
+                        elif suffix == "post":
+                            platforms[platform_name].plugins[
+                                plugin_name
+                            ].custom_docs_post = final_markdown
                         else:
-                            create_or_update(
-                                source_documentation,
-                                [
-                                    platform_name,
-                                    "plugins",
-                                    file_name,
-                                    "custom_docs_post",
-                                ],
-                                final_markdown,
+                            raise ValueError(
+                                f"{file_name} needs to be of the form <plugin>_pre.md or <plugin>_post.md"
                             )
+
+                    else:  # assume this is the platform post.
```

Comment:
Maybe we should let it fail instead of making guesses. 
Alternatively, we could log a warning to catch these cases if they occur.

Reply:
I would agree - I kept this as-is (only added a comment) to keep the scope of this PR small